plugins/org.yakindu.sct.generator.c/src/org/yakindu/sct/generator/c/GeneratorPredicate.xtend

```diff
@@ -28,4 +28,12 @@ class GeneratorPredicate extends org.yakindu.sct.generator.core.extensions.Gener
 	def useOutEventObservables() {
 		entries.getOutEventObservablesUsed(entry)
 	}
+	
+	def useOutEventGetters() {
+		entries.getOutEventGettersUsed(entry)
+	}
+	
+	def needsClearOutEventsFunction() {
+		!useOutEventObservables || useOutEventGetters
```

Comment:
simply: return useOutEventGetters 
Observables have no impact on getters. If we require that getters should be used if seen specifies "observables = false" and defines nothing for getters  then we should implement this in useOutEventGetters accordingly.